src/advanced_inv.cpp

```diff
@@ -631,7 +648,12 @@ void advanced_inv_area::init()
     static const std::array<ter_id, 6> ter_water = {
         {t_water_dp, t_water_pool, t_swater_dp, t_water_sh, t_swater_sh, t_sewage}
     };
-    if(is_any_of(g->m.ter(pos), ter_water)) {
+    const auto &map = g->m;
+    auto ter_check = [&map, this]
```

Comment:
- if you only need `pos` inside the lambda, you should only capture `pos` and not `this`.
- `g->m` is global, no need to capture it, it can be used directly in the lambda.